Adding Getting Started (#11)
xmidt-org/docs: content/docs/operating/troubleshooting.md

> ---

Reviewer: Forgot to put this in my review, but I think this page looks great!! 💯

title: Troubleshooting
sort_rank: 2
---

# Troubleshooting

## Device Errors

### Device is not showing up in Cluster/Talaria

- Problem: Parados has the incorrect url.

>   - Solution: Validate Parados has the correct url.

Reviewer: I think this is the first time I've read Parados anywhere? Should this say something like `mock CPE/device` instead? Or add Parados to other pages where they are talking about mock devices.

    ```bash
    docker run -e URL=http://HOSTNAME:PRIMARY_PORT rdkb-simulator
    # HOSTNAME should be either a talaria or petasos address, with
    # the correct corresponding PRIMARY_PORT
    ```

- Problem: TLS is inconsistent between talaria and petasos.

>   - Fixed Solution: Change talaria and petasos Configuration to be consistent with `http` or `https`.

Reviewer: ```suggestion
  - Fixed Solution: Change talaria and petasos configurations to be consistent with `http` or `https`.
```

    ```yaml
    # talaria  
    service:
      defaultScheme: http
      fixed:

>         - http://talaria:6200

Reviewer: This example has fixed values but the consul one has variables.  Seems like we're moving towards variables, so maybe add them here?

    # petasos
    service:
    defaultScheme: http
    fixed:
      - http://talaria:6200
    ```

  - Consul Solution: Change talaria and petasos Configuration to be consistent with `http` or `https`. `https` is default.

    ```yaml
    # talaria  
    service:
      defaultScheme: http
      # ^ important
      consul:
        client:

>           address: "CONSUL_ADDRESS:8500"

Reviewer: Can we explain the variables in these two configurations?
- [ ] `CONSUL_ADDRESS`
- [ ] `HOSTNAME`
- [ ] `HEALTH_PORT`

          scheme: "http"
          waitTime: "30s"
        disableGenerateID: true
        vnodeCount: 211
        watches:
          -
            service: "talaria"
            tags:
              - "dev"
              - "docker"
            passingOnly: true
        registrations:
          -
            id: "HOSTNAME"
            name: "talaria"
            tags:
              - "dev"
              - "docker"
              - "stage=dev"
              - "flavor=docker"

>             address: "http://HOSTNAME"

Reviewer: this doesn't need a port?

            # ^ important
            scheme: "http"
            # ^ important
            port: 6210
            checks:
              -
                checkID: "talaria-1:http"
                http: "http://HOSTNAME:HEALTH_PORT/health"
                interval: "30s"
                deregisterCriticalServiceAfter: "70s"

    # petasos
    service:
    defaultScheme: http
    # ^ important
    consul:
      client:
        address: "consul0:8500"
        scheme: "http"
        waitTime: "30s"
      disableGenerateID: true
      watches:
        -
          service: "talaria"
          passingOnly: true
      vnodeCount: 211
    ```

- Problem: Talaria is not registering with Consul.

>   - Detection: On a consul node run the following commands.

Reviewer: ```suggestion
  - Detection: On a consul node, run the following command(s):
```

    ```bash
    consul catalog services
    # talaria should show up in list, if not try the solution
    ```

>   - Solution: Validate Consul client config, with TLS and address

Reviewer: ```suggestion
  - Solution: Validate Consul client config, with TLS and address.
```

    ```yaml
    consul:
      client:

>         address: "HOSTNAME:API_PORT"

Reviewer: Explain variables?
- [ ] `HOSTNAME`
- [ ] `API_PORT`

>   - Solution: Validate Consul client config, with TLS and address.

Reviewer: ```suggestion
  - Solution: Validate Consul client block in Talaria config, with TLS and address.
```

        # ^ important
        scheme: "http"
        # ^ important
        waitTime: "30s"
    ```

## Event Errors
- Problem: Caduceus is not receiving events.
  - Solution: Update the talaria yaml with the correct information.

    ```yaml
    eventMap:

>         default: http://CADUCEUS_HOSTNAME:PRIMARY_PORT/api/v3/notify

Reviewer: Explain variables?
- [ ] `CADUCEUS_HOSTNAME`
- [ ] `PRIMARY_PORT`

          # double check http vs https and the hostname
    ```

> # General Error

Reviewer: ```suggestion
# General Errors
```

- Problem: Docker networking with localhost.
  - Solution: Use the ip address of the host machine.

>   You can not use localhost when working with docker.

Reviewer: ```suggestion
  You can not use `localhost` when working with docker.
```
